skia/RSkComponentProviderRNSVG.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 1994-2023 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "react/renderer/components/rnsvg/RNSVGComponentDescriptors.h"
+
+#include "ReactSkia/components/RSkComponentProvider.h"
+#include "ReactSkia/utils/RnsUtils.h"
+#include "RSkComponentRNSVGCircle.h"
+#include "RSkComponentRNSVGDefs.h"
+#include "RSkComponentRNSVGEllipse.h"
+#include "RSkComponentRNSVGGroup.h"
+#include "RSkComponentRNSVGLine.h"
+#include "RSkComponentRNSVGPath.h"
+#include "RSkComponentRNSVGRect.h"
+#include "RSkComponentRNSVGUse.h"
+#include "RSkComponentRNSVGSvgView.h"
+
+#define SETUP_COMPOMNENT_PROVIDER_FOR(SVGElement) \
+class RSkComponentProvider##SVGElement : public RSkComponentProvider { \
+ public: \
+  RSkComponentProvider##SVGElement(){} \
+  ComponentDescriptorProvider GetDescriptorProvider() override { \
+    return concreteComponentDescriptorProvider<SVGElement##ComponentDescriptor>(); \
+  }\
+  std::shared_ptr<RSkComponent> CreateComponent( \
+    const ShadowView &shadowView) override { \
```

Comment:
nice approach of handling multiple providers !
probably we can use the same in RNS as well :)